core/validation/checks.py

```diff
@@ -0,0 +1,131 @@
+from abc import ABC, abstractmethod
+
+import pandas as pd
+
+
+class Check(ABC):
+    """
+    Abstract base class representing a validation check on a specific tab/cell of a workbook.
+
+    Attributes:
+        sheet (str): The name of the sheet in the workbook.
+        column (int): The column index of the cell to check.
+        row (int): The row index of the cell to check.
+        expected_values (tuple or None): A tuple of expected values for the cell, or None if any value is acceptable.
+        error_message (str): The error message sent to the user if the check fails.
+        actual_value (str or None): The actual value retrieved from the cell on the ingested spreadsheet.
+
+    Methods:
+        run(workbook: dict[str, pd.DataFrame]) -> bool:
+            Execute the check on the provided workbook.
+    """
+
+    def __init__(self, sheet: str, column: int, row: int, expected_values: tuple | None, error_message: str):
+        self.sheet = sheet
+        self.column = column
+        self.row = row
+        self.expected_values = expected_values
+        self.error_message = error_message
+        self.actual_value = None
+
+    def run(self, workbook: dict[str, pd.DataFrame]) -> bool:
+        self.actual_value = str(workbook[self.sheet].iloc[self.column][self.row]).strip()
+        if self.actual_value not in self.expected_values:
+            return False
+        return True
+
+
+class BasicCheck(Check):
+    """
+    Used for checks that require no extra parameters or variable error messages
+    """
+
+    pass
+
+
+class DynamicCheck(Check, ABC):
+    """
+    Used to represent a dynamic check, i.e. containing parameters dynamically generated at runtime. Contains methods for
+    calculating the expected values on a cell, and substituting placeholders in error messages with the calculated
+    parameters.
+
+    Attributes:
+        _dynamic_params (dict): Dictionary containing values used to calculate the expected values dynamically.
+        _error_message_with_placeholders (str): The error message template before substitution.
+
+    """
+
+    _dynamic_params: dict  # Values used to calculate the expected values
+    _error_message_with_placeholders: str  # The error message before substitution
+
+    def __init__(self, sheet: str, column: int, row: int, dynamic_params: dict, error_message_with_placeholders: str):
+        super().__init__(sheet, column, row, None, None)
+        self._dynamic_params = dynamic_params
+        self._error_message_with_placeholders = error_message_with_placeholders
+
+    @abstractmethod
+    def calculate_expected_values(self) -> tuple:
+        pass
```

Comment:
One alternative could be for this `ABC` to just contain this `abstractmethod` and then the children override `run` and init any additional instance variables they need. `AuthorisationCheck` could then implement `substitute_error_message` without sharing it with `ConflictingCheck` and we wouldn't need `_dynamic_params` either